lib/improver/tests/spotdata_new/spotdata_new/test_SpotExtraction.py

```diff
@@ -0,0 +1,259 @@
+    cube for a given neighbour_selection_method."""
+
+    def test_nearest(self):
+        """Test extraction of nearest neighbour x and y indices."""
+        plugin = SpotExtraction(neighbour_selection_method='nearest')
+        expected = self.neighbours[:, 0, 0:2].astype(int)
+        result = plugin.extract_coordinates(self.neighbour_cube)
+        self.assertArrayEqual(result.data, expected)
+
+    def test_nearest_land(self):
+        """Test extraction of nearest land neighbour x and y indices."""
+        plugin = SpotExtraction(neighbour_selection_method='nearest_land')
+        expected = self.neighbours[:, 1, 0:2].astype(int)
+        result = plugin.extract_coordinates(self.neighbour_cube)
+        self.assertArrayEqual(result.data, expected)
+
+    def test_invalid_method(self):
+        """Test attempt to extract neighbours found with a method that is not
+        available within the neighbour cube. Raises an exception."""
+        plugin = SpotExtraction(neighbour_selection_method='furthest')
+        msg = 'The requested neighbour_selection_method "furthest" is not'
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin.extract_coordinates(self.neighbour_cube)
+
+
+class Test_extract_diagnostic_data(Test_SpotExtraction):
+
+    """Test the extraction of data from the provided coordinates."""
+
+    def test_xy_ordered_cube(self):
+        """Test extraction of diagnostic data that is natively ordered xy."""
+        plugin = SpotExtraction()
+        expected = [0, 0, 12, 12]
+        result = plugin.extract_diagnostic_data(self.coordinate_cube,
+                                                self.diagnostic_cube_xy)
+        self.assertArrayEqual(result, expected)
+
+    def test_yx_ordered_cube(self):
+        """Test extraction of diagnostic data that is natively ordered yx.
+        This will be reordered before extraction to become xy."""
+        plugin = SpotExtraction()
+        expected = [0, 0, 12, 12]
+        result = plugin.extract_diagnostic_data(self.coordinate_cube,
+                                                self.diagnostic_cube_yx)
+        self.assertArrayEqual(result, expected)
+
+
+class Test_build_diagnostic_cube(Test_SpotExtraction):
+
+    """Test the building of a spot data cube with given inputs."""
+
+    def test_building_cube(self):
+        """Test that a cube is built as expected."""
+        plugin = SpotExtraction()
+        spot_values = [0, 0, 12, 12]
+        result = plugin.build_diagnostic_cube(self.neighbour_cube,
+                                              self.diagnostic_cube_xy,
+                                              spot_values)
+        self.assertArrayEqual(result.coord('latitude').points, self.latitudes)
+        self.assertArrayEqual(result.coord('longitude').points,
+                              self.longitudes)
+        self.assertArrayEqual(result.data, spot_values)
+
+
+class Test_process(Test_SpotExtraction):
```

Comment:
Do you have a test anywhere checking the carrying forward of the extra dimensions (like realization). I know it works but we probably need a test somewhere.

Reply:
Added a case with a leading realization coordinate and check that the same coordinate is present on the resulting cube.